src/modules/common/mentors.service.ts

```diff
@@ -22,7 +22,19 @@ export class MentorsService {
     const onlyMentors: any = {
       roles: 'Mentor',
     };
-    const projections = this.getMentorFields(isLoggedIn);
+    const projections = this.getMentorFields();
+
+    if (isLoggedIn) {
+      // Filter by availability
+      if (filters.available !== undefined) {
```

Comment:
```suggestion
      if (filters.available) {
        onlyMentors.available = filters.available;
        projections.channels = filters.available;
      }
        
```

Reply:
Tried that first, mongo complaints about mixing projections 🤷‍♂️